src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeReprocessService.java

```diff
@@ -28,5 +33,35 @@ public void reprocessEnvelope(String envelopeId) {
             .orElseThrow(
                 () -> new EnvelopeNotFoundException("Envelope with id " + envelopeId + " not found")
             );
+
+        validateEnvelopeState(envelope);
+
+        createEvent(envelope);
+
+        envelope.setStatus(Status.UPLOADED);
+        envelopeRepository.save(envelope);
+    }
+
+    private void createEvent(Envelope envelope) {
+        ProcessEvent event = new ProcessEvent(
+            envelope.getContainer(),
+            envelope.getZipFileName(),
+            Event.ONLINE_STATUS_CHANGE
+        );
+        event.setReason("Moved to UPLOADED status to reprocess the envelope");
+        processEventRepository.save(event);
+    }
+
+    private void validateEnvelopeState(Envelope envelope) {
+        if (envelope.getCcdId() != null) {
+            throw new EnvelopeProcessedException(
+                "Envelope with id " + envelope.getId() + " has already been processed"
+            );
+        }
+        if (envelope.getStatus() == Status.UPLOADED) {
```

Comment:
if envelope is in uploaded status does not mean system finish processing ?

Reply:
Finish means COMPLETED status

Reply:
I think, system might be still trying to process in UPLOADED status?

Reply:
If it is in UPLOADED status - it is being processed